spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/AddFilesProcedure.java

```diff
@@ -196,11 +193,36 @@ private void importFileTable(
       String format,
       Map<String, String> partitionFilter,
       boolean checkDuplicateFiles,
-      PartitionSpec spec,
       int parallelism) {
+
+    org.apache.spark.sql.execution.datasources.PartitionSpec inferredSpec =
+        getInferredSpec(spark(), tableLocation);
+
+    List<String> sparkPartNames =
+        JavaConverters.seqAsJavaList(inferredSpec.partitionColumns()).stream()
+            .map(StructField::name)
+            .map(name -> name.toLowerCase(Locale.ROOT))
+            .collect(Collectors.toList());
+    PartitionSpec matchingSpec = null;
+    for (PartitionSpec icebergSpec : table.specs().values()) {
+      List<String> icebergPartNames =
+          icebergSpec.fields().stream()
+              .map(PartitionField::name)
+              .map(name -> name.toLowerCase(Locale.ROOT))
+              .collect(Collectors.toList());
+      if (icebergPartNames.equals(sparkPartNames)) {
+        matchingSpec = icebergSpec;
+        break;
+      }
+    }
+    if (matchingSpec == null) {
+      // If the inferred spec does not match any existing spec, use the Iceberg table's latest spec
+      matchingSpec = table.spec();
```

Comment:
Taking the table spec if we don't find any matches.
Should we throw an error otherwise?

Reply:
I think this is an error case

Reply:
Using the findCompatibleSpec method from SparkTableUtil. so now this will throw an error